src/components/filters/save-filter-criteria/index.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright 2019 OpenStack Foundation
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+import React, { useState } from 'react'
+import T from 'i18n-react/dist/i18n-react';
+import { Input, RadioList } from 'openstack-uicore-foundation/lib/components'
+import { hasErrors } from "../../../utils/methods";
+
+import styles from './index.module.less'
+
+const SaveFilterCriteria = ({ onSave }) => {
+
+    const [customName, setCustomName] = useState('');
+    const [visibility, setVisibility] = useState(null);
+    const [errors, setErrors] = useState({});
+
+    const visibility_options = [
+        { value: 'Me', label: 'Me' },
```

Comment:
for value use constants
define a constant on an util file ( eg FilterCriteriaConstants.js)
for label please resource it on i18n file